tests: Bluetooth: tester: Add initial support for HAP

MariuszSkamra opened the pull request:
This adds initial support for Hearing Aid Profile BTP service commands.

MariuszSkamra committed:
tests: Bluetooth: tester: Add initial support for HAP

This adds initial support for Hearing Aid Profile BTP service commands.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

Thalley commented on subsys/bluetooth/audio/has_client.c:
```diff
@@ -1000,3 +1000,24 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	.disconnected = disconnected,
 };
+
+#if defined(CONFIG_BT_TESTING)
+int bt_has_client_test_info_get(struct bt_conn *conn, struct bt_has_client_test_info *info)
+{
+	struct has_inst *inst;
+
+	__ASSERT_NO_MSG(conn != NULL);
+	__ASSERT_NO_MSG(info != NULL);
+
+	inst = inst_by_conn(conn);
+	if (inst == NULL) {
+		return -ENODEV;
+	}
+
+	info->has_hearing_aid_features_handle = inst->features_subscription.value_handle;
+	info->has_control_point_handle = inst->control_point_subscription.value_handle;
+	info->has_active_preset_index_handle = inst->active_index_subscription.value_handle;
+
+	return 0;
+}
+#endif /* CONFIG_BT_TESTING */
```
What we have done for the other implementations is to move the definition of e.g. `struct has_inst` to the internal header files, and simply included them in the tests. 

This avoids having test specific functions in the implementations. 

Did you consider this approach, and if so, why did you go with this approach?

MariuszSkamra replied:
No, I did not consider this, but it's a better approach actually. Will do that.

Thalley commented on subsys/bluetooth/audio/has_internal.h:
```diff
@@ -8,6 +8,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/bluetooth/bluetooth.h>
```
Was this to have the `struct bt_conn`? If so, shouldn't this have been `conn.h`?

Thalley commented on tests/bluetooth/tester/src/btp_hap.c:
```diff
@@ -0,0 +1,292 @@
+	LOG_DBG("conn %p err %d", (void *)conn, err);
+
+	if (err != 0 && err != BT_ATT_ERR_ATTRIBUTE_NOT_FOUND) {
+		LOG_DBG("Discover remote ASEs failed: %d", err);
```
This isn't discovering ASEs, right?

MariuszSkamra committed:
Bluetooth: audio: has_client: Move bt_has_client structure to header

This moves bt_has_client instance structure to has_internal.h header
file to be able to access to the instance members in tests.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>
